internal/controllers/clusterextension_controller.go

```diff
@@ -306,12 +323,34 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 		},
 	}
 
-	rel, state, err := r.getReleaseState(ac, ext, chrt, values, post)
+	rel, desiredRel, state, err := r.getReleaseState(ac, ext, chrt, values, post)
 	if err != nil {
 		setInstalledStatusConditionFailed(ext, fmt.Sprintf("%s:%v", ocv1alpha1.ReasonErrorGettingReleaseState, err))
 		return ctrl.Result{}, err
 	}
 
+	for _, preflight := range r.Preflights {
+		if _, ok := preflight.(*crdupgradesafety.Preflight); ok && ext.Spec.Preflight.CRDUpgradeSafety.Disabled {
+			// Skip this preflight check because it is of type *crdupgradesafety.Preflight and the CRD Upgrade Safety
+			// preflight check has been disabled
+			continue
+		}
```

Comment:
I did some investigation into the CI failures and it looks like it was due to a nil pointer reference. I forgot in my initial recommendation that the `Preflight` and `CRDUpgradeSafety` fields are pointers so I think this should actually be:
```go
if ext.Spec.Preflight != nil {
	if ext.Spec.Preflight.CRDUpgradeSafety != nil {
		if _, ok := preflight.(*crdupgradesafety.Preflight); ok && ext.Spec.Preflight.CRDUpgradeSafety.Disabled {
			// Skip this preflight check because it is of type *crdupgradesafety.Preflight and the CRD Upgrade Safety
			// preflight check has been disabled
			continue
		}
	}
}
```

Reply:
Thanks :grinning: updated the PR